rules/aip0132/resource_reference_type.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0132
+
+import (
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// List methods should reference the target resource via `child_type` or the
+// parent directly via `type`.
+var resourceReferenceType = &lint.MethodRule{
+	Name: lint.NewRuleName(132, "resource-reference-type"),
+	OnlyIf: func(m *desc.MethodDescriptor) bool {
+		repeated := utils.GetRepeatedMessageFields(m.GetOutputType())
+		p := m.GetInputType().FindFieldByName("parent")
+		return isListMethod(m) && p != nil && utils.GetResourceReference(p) != nil && len(repeated) > 0
+	},
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// The first repeated message field must be the paginated resource.
+		repeated := utils.GetRepeatedMessageFields(m.GetOutputType())
+		resource := utils.GetResource(repeated[0].GetMessageType())
```

Comment:
Same as above and below: `repeated[0]` relies on the implication that the first repeated field is the resource. Do we lint for this, explicitly?